viewing_party/main.py

```diff
@@ -0,0 +1,113 @@
+def create_movie(movie_title, genre, rating):
+
+    movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+    for info in movie:
+        if movie[info] == None:
+            return None
+
+    return movie
+
+def add_to_watched(user_data, movie):
+
+  user_data.update({"watched": [movie]})
+  return user_data
+
+def add_to_watchlist(user_data, movie):
+
+  user_data.update({"watchlist":[movie]})
+  return user_data  
+
+def watch_movie(user_data, movie):
+
+    for title in user_data["watchlist"]:
+        if title["title"] == movie:
+            user_data["watched"].append(title)
+            user_data["watchlist"].remove(title)
+
+    return user_data
+
+def get_watched_avg_rating(user_data):
+
+    ratings = []
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    for movie in user_data["watched"]:
+        ratings.append(movie["rating"])
+    
+    return sum(ratings)/len(ratings)
+
+def get_most_watched_genre(user_data):
+
+    genres = []
+    if len(user_data["watched"]) == 0:
+        return None
+    for movie in user_data["watched"]:
+      for genre in movie["genre"]:
+        genres.append(movie["genre"])
+        
+    return max(set(genres), key=genres.count)
+
+def get_friends_movies(user_data):
+
+    friends_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movies.append(movie)
+
+    return friends_movies
+
+def get_unique_watched(user_data):
+    
+    unique_movies = []
```

Comment:
Later on in Ada, I'm guessing you'll learn about another data structure called the `set`. This cool structure can't have any duplicates, so instead of the conditional statement to weed out any dupes, the set would do the work for you.

Reply:
Dicts are unhashable though, so you can't make sets with them :(